extensions-core/datasketches/src/main/java/org/apache/druid/query/aggregation/datasketches/hll/HllSketchHolderObjectStrategy.java

```diff
@@ -48,12 +49,18 @@ public int compare(final HllSketchHolder sketch1, final HllSketchHolder sketch2)
   @Override
   public HllSketchHolder fromByteBuffer(final ByteBuffer buf, final int size)
   {
+    if (size == 0 || isSafeToConvertToNullSketch(buf)) {
```

Comment:
If the `size` parameter is less than 8, then I believe that the `isSafeToConvertToNullSketch` method could end up throwing an exception when it reads out the value for `curMode`.  It would likely make sense to pass the size parameter into the `isSafe...` method and check the size there as well, returning false if it's too small.

Reply:
Added a check on size inside the method wherever necessary